text/0000-build-std/4-build-std-always.md

```diff
@@ -0,0 +1,1257 @@
+
+## Caching
+[caching]: #caching
+
+Standard library artifacts built by build-std will not be shared between crates
+or workspaces, as they only exist in Cargo's target directory for a specific
+crate or workspace ([?][rationale-caching]).
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not globally cache builds of the standard library?*][rationale-caching]
+
+## Generated documentation
+[generated-documentation]: #generated-documentation
+
+When running `cargo doc` for a project to generate documentation and rebuilding
+the standard library, the generated documentation for the user's crates will
+link to the locally generated documentation for the `core`, `alloc` and `std`
+crates, rather than the upstream hosted generation as is typical for non-locally
+built standard libraries.
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not link to hosted standard library documentation in generated docs?*][rationale-generated-docs]
+
+## Cargo subcommands
+[cargo-subcommands]: #cargo-subcommands
+
+As opaque dependencies, any Cargo command which accepts a package spec with `-p`
+will only additionally recognise `core`, `alloc` and `std` and none of their
+dependencies. Many of Cargo's subcommands will need modification to support
+build-std:
+
+[`cargo clean`][cargo-clean] will additionally delete any builds of the standard
+library performed by build-std.
+
+[`cargo fetch`][cargo-fetch] will not fetch the standard library dependencies as
+they are already vendored in the `rust-src` component.
+
+[`cargo miri`][cargo-miri] is not built into Cargo, it is shipped by miri, but
+is mentioned in Cargo's documentation. `cargo miri` is unchanged by this RFC,
+but build-std is one step towards `cargo miri` requiring less special support.
+
+> [!NOTE]
+>
+> `cargo miri` could be re-implemented using build-std to enable a `miri`
+> profile and always rebuild. The `miri` profile would be configured in the
+> standard library's workspace, setting the flags/options necessary for `miri`.
+
+[`cargo report`][cargo-report] will not include reports from the standard
+library crates or their dependencies.
+
+[`cargo update`][cargo-update] will not update the dependencies of `std`,
+`alloc` and `core`, as these are vendored as part of the distribution of
+`rust-src` and resolved separately from the user's dependencies. Neither will
+`std`, `alloc` or `core` be updated, as these are unversioned and always match
+the current toolchain version.
+
+[`cargo vendor`][cargo-vendor] will not vendor standard library dependencies.
+Vendoring these and using them later would effectively pin the crate to the
+version of the language and toolchain used when vendoring was performed (as the
+vendored standard library source would only work with that toolchain version).
+Standard library crates are already vendored in the `rust-src` component, so do
+not require network access once downloaded.
```

Comment:
```suggestion
[`cargo vendor`][cargo-vendor] will not vendor the standard library crates or their dependencies.
These are pre-vendored as part of the `rust-src` component ([?][rationale-vendoring]).
```

IMO this is simpler and avoids ambiguity around what exactly is in the vendored sources.

Reply:
Adopted this wording in 8f82df45